src/testing/cluster.zig

```diff
@@ -623,14 +623,21 @@ pub fn ClusterType(comptime StateMachineType: anytype) type {
             const result = reformat.done() orelse return;
             assert(result == .ok);
 
-            const releases = cluster.replica_health[replica_index].reformatting.releases;
             reformat.deinit(cluster.allocator);
             cluster.replica_reformats[replica_index] = null;
             cluster.replica_health[replica_index] = .down;
-            cluster.replica_restart(replica_index, &releases) catch unreachable;
+            cluster.replica_restart(replica_index) catch unreachable;
             cluster.state_checker.reformat(replica_index);
         }
 
+        pub fn replica_set_releases(
+            cluster: *Cluster,
+            replica_index: u8,
+            releases: *const vsr.ReleaseList,
+        ) void {
```

Comment:
I was initially thinking that we could assert that the release list contains the current release.
But we should be capable of testing even when that isn't the case -- in that case, replica should not advertise/upgrade.